ra-trustvox/components/TrustvoxCertificate.tsx

```diff
@@ -0,0 +1,20 @@
+import { state } from "../mod.ts";
+
+export default function TrustvoxCertificate() {
+  const { enableStaging = false } = state;
```

Comment:
I think you're misusing the state, you should not use state `shared` across all app installations otherwise you can have many different types of issues related to concurrency and/or previewing app using different props

Reply:
Instead, you need to:

Transform this into a section and add a `inline loader` in order to access `AppContext` and then accessing the state from `ctx` there are multiple examples of how to do it in this repository, you can easily find one of them

Reply:
This is an example, the stars component is inserted within the product card component, already existing in the customer's store.

![image](https://github.com/user-attachments/assets/5ba0e0ae-0789-4a94-9530-336ce21c0104)

To use it, the client must import it into the code:

```
import TrustvoxProductDetailsRate from "apps/ra-trustvox/components/TrustvoxProductDetailsRate.tsx";
```

and render wherever you want:

```
<TrustvoxProductDetailsRate productId={product.productID} />
```

For these cases, would you have any idea how I can do it, adapt it to the way you suggested? @mcandeia

Reply:
Components should be self-contained and should not depend on external data without receiving from props or from context. For instance, you don't have access to ctx inside an island because it is a server-side only data. So you need to drilldown props until it reaches your components, so you either need sections or you need your component to receive the state you need

Reply:
So how can I solve my problem? Considering these cases, will the user import the component directly into the code? I still don't understand how to do this